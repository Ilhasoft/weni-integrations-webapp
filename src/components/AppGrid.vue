<template>
  <div>
    <section id="app-grid">
      <p class="app-grid__title">{{ $t(section) }}</p>

      <div class="app-grid__content">
        <unnnic-card
          ref="unnnic-marketplace-card"
          class="app-grid__content__item"
          v-for="app in apps"
          v-bind:key="app.id"
          type="marketplace"
          :title="app.name"
          :description="app.description"
          :icon="app.icon"
          :id="app.id"
          :comments="`${app.comments_count} ${$t('comments')}`"
          :rating="app.rating.average || 0"
          :iconSrc="app.icon"
          :typeAction="type"
          :buttonAction="/* istanbul ignore next */ () => openAddModal(app.id, app.name)"
          clickable
          @openModal="openAppModal(app.code)"
        />
      </div>
    </section>

    <unnnic-modal
      ref="unnnic-add-modal"
      :showModal="showAddModal"
      :text="$t('installed_app_title')"
      scheme="feedback-green"
      modal-icon="check-circle-1-1"
      @close="closeAddModal"
    >
      <span slot="message" v-html="$t('installed_app_description')"></span>
      <unnnic-button
        ref="unnnic-add-modal-close-button"
        slot="options"
        type="terciary"
        @click="closeAddModal"
        >{{ $t('close') }}</unnnic-button
      >
      <unnnic-button
        ref="unnnic-add-modal-navigate-button"
        slot="options"
        type="primary"
        @click="navigateToMyApps()"
      >
        {{ $t('access_my_apps') }}
      </unnnic-button>
    </unnnic-modal>
  </div>
</template>

<script>
  import { mapActions } from 'vuex';
  export default {
    name: 'AppGrid',
    props: {
      section: {
        type: String,
        default: null,
        validator(value) {
          return (
            [
              'communication_channels',
              'attendance_managers',
              'configured_apps',
              'installed_apps',
            ].indexOf(value) !== -1
          );
        },
      },
      type: {
        type: String,
        validator(value) {
          return ['add', 'config', 'edit'].indexOf(value) !== -1;
        },
      },
    },
    data() {
      return {
        showAddModal: false,
        apps: [],
      };
    },
    async mounted() {
      const category = this.getCategory(this.section);
      this.loadApps({ category });
    },
    methods: {
      ...mapActions(['getAllAppTypes']),
      async loadApps(filter) {
        if (this.type == 'add') {
          const { data } = await this.getAllAppTypes(filter);
          this.apps = data;
        } else {
          this.apps = [
            {
              code: 'wwc',
              name: 'Weni Web Chat MOCK',
              description:
                'Ullamco occaecat et id cillum. Amet exercitation nisi amet fugiat mollit minim est. Officia in enim amet ipsum Lorem velit sint pariatur sunt magna cupidatat. Magna non ea qui nisi ut.s',
              summary: 'Ullamco occaecat et id cillum.',
              category: 'channel',
              bg_color: '#00DED333',
              icon: 'https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg',
              assets: [
                {
                  type: 'image_banner',
                  url: 'https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/Ilustra%C3%A7%C3%A3o%20banner%201.png',
                  description: '',
                },
              ],
              usersCount: 25,
              rating: {
                average: 4.9,
              },
              commentsCount: 3,
              config: {
                title: 'Chat Title',
                subtitle: 'Chat subtitle',
                placeholder: 'Chat placeholder',
                fullScreenButton: false,
                unreadIndicator: false,
                keepHistory: false,
                avatarImage:
                  'https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/iconfinder_13-whatsapp_4202050+1.svg',
                customCSS: '',
                primaryColor: '#009E96',
                messageDelay: 1,
              },
            },
          ];
        }
      },
      getCategory(section) {
        const categories = {
          communication_channels: 'channel',
          attendance_managers: 'ticket',
          configured_apps: null,
          installed_apps: null,
        };
        return categories[section];
      },
      openAddModal() {
        this.showAddModal = true;
      },
      closeAddModal() {
        this.showAddModal = false;
      },
      navigateToMyApps() {
        this.$router.push('apps');
      },
      openAppModal(code) {
        this.$router.push(`/apps/${code}/details`);
      },
    },
  };
</script>

<style lang="scss" scoped>
  .app-grid {
    &__title {
      color: $unnnic-color-neutral-darkest;
      font-size: $unnnic-font-size-title-sm;
    }

    &__content {
      display: grid;
      grid-template-columns: repeat(auto-fit, minmax(256px, 256px));
      grid-gap: $unnnic-spacing-stack-sm;
      align-items: flex-start;
    }
  }
</style>
