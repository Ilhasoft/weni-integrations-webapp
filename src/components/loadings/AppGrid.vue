<template>
  <div>
    <div class="title">
      <unnnic-skeleton-loading tag="div" width="40px" height="40px" />
      <unnnic-skeleton-loading tag="div" width="208px" height="40px" />
    </div>
    <div class="grid">
      <unnnic-skeleton-loading
        v-for="index in 4"
        :key="index"
        class="grid__item"
        tag="div"
        width="256px"
        height="200px"
      />
    </div>
  </div>
</template>

<script>
  // import SkeletonLoading from '../Skeleton/SkeletonLoading.vue';
  export default {};
</script>

<style lang="scss" scoped>
  .title {
    display: flex;
    gap: $unnnic-spacing-inline-xs;
    margin: $unnnic-spacing-stack-md 0;
  }
  .grid {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(256px, 256px));
    grid-gap: $unnnic-spacing-stack-sm;
    align-items: flex-start;
  }
</style>
