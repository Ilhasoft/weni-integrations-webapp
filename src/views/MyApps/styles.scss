.my-apps {
    &__search {
      margin-top: $unnnic-spacing-stack-sm;
      margin-bottom: $unnnic-spacing-stack-lg;

      &__results {
        font-size: $unnnic-font-size-body-lg;
        line-height: $unnnic-font-size-body-lg + $unnnic-line-height-md;
        color: $unnnic-color-neutral-darkest;

        &__highlight {
          font-weight: $unnnic-font-weight-bold;
        }
      }
    }

    &__empty {
      display: flex;
      flex-direction: column;
      align-items: center;
      margin-top: $unnnic-spacing-stack-xl;

      &__description {
        &__main {
          font-family: $unnnic-font-family-secondary;
          font-weight: $unnnic-font-weight-bold;
          font-size: $unnnic-font-size-title-sm;
          line-height: $unnnic-line-height-md + $unnnic-font-size-title-sm;
          color: $unnnic-color-neutral-dark;
          text-align: center;
        }

        &__secondary {
          display: flex;
          gap: $unnnic-inline-nano;
          text-align: center;

          font-family: $unnnic-font-family-secondary;
          font-size: $unnnic-font-size-body-lg;
          line-height: $unnnic-line-height-md + $unnnic-font-size-body-lg;
          color: $unnnic-color-neutral-cloudy;

          &__link {
            cursor: pointer;
            font-weight: $unnnic-font-weight-bold;
            border-bottom: $unnnic-border-width-thinner solid $unnnic-color-neutral-cloudy;
          }
        }
      }
    }

    &__sections {
      display: flex;
      flex-direction: column;
      gap: $unnnic-spacing-stack-lg;

      margin-bottom: $unnnic-spacing-stack-md;

      &__grids {
        display: flex;
        flex-direction: column;
        gap: $unnnic-spacing-stack-lg;
      }
    }
  }