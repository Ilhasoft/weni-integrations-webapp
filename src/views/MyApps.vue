<template>
  <div>
    <app-grid section="configured" type="edit" />
    <app-grid section="installed" type="config" />
  </div>
</template>

<script>
  import AppGrid from '../components/AppGrid.vue';
  export default {
    name: 'Apps',
    components: { AppGrid },
  };
</script>

<style scoped></style>
