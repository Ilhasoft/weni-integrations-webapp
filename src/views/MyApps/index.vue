<template>
  <div class="my-apps">
    <unnnic-autocomplete
      v-if="hasApps"
      v-model="searchTerm"
      class="my-apps__search"
      :placeholder="$t('apps.discovery.search.placeholder')"
      icon-left="search-1"
      :data="searchOptions"
      id="search"
    />

    <span
      v-if="hasApps && searchTerm && searchTerm.trim()"
      class="discovery-content__search__results"
    >
      {{ $t('apps.discovery.search.results') }}
      <span class="discovery-content__search__results__highlight">
        {{ `“${searchTerm}”...` }}
      </span>
    </span>

    <div v-if="hasApps" class="my-apps__sections">
      <div v-if="searchTerm && !filteredConfiguredApps.length && !filteredInstalledApps.length">
        <EmptyApps :term="searchTerm" @clear="navigateToDiscovery" />
      </div>
      <div class="my-apps__sections__grids" v-else>
        <app-grid
          section="configured"
          type="edit"
          :loading="loadingConfiguredApps"
          :apps="filteredConfiguredApps"
          @update="fetchConfigured"
        />
        <app-grid
          section="installed"
          type="config"
          :loading="loadingInstalledApps"
          :apps="filteredInstalledApps"
          @update="fetchInstalled"
        />
      </div>
    </div>
    <div v-else class="my-apps__empty">
      <img class="my-apps__empty__image" src="@/assets/svgs/empty-apps.svg" />
      <div class="my-apps__empty__description__main">
        {{ $t('apps.myApps.empty.no-integrations') }}
      </div>
      <div class="my-apps__empty__description__secondary">
        <div>{{ $t('apps.myApps.empty.search') }}</div>
        <div class="my-apps__empty__description__secondary__link" @click="navigateToDiscovery()">
          {{ $t('apps.myApps.empty.link') }}
        </div>
      </div>
    </div>
  </div>
</template>

<script>
  import AppGrid from '@/components/AppGrid/index.vue';
  import EmptyApps from '@/components/EmptyApps/index.vue';
  import { mapState, mapActions } from 'pinia';
  import unnnic from '@weni/unnnic-system';
  import { auth_store } from '@/stores/modules/auth.store';
  import { my_apps } from '@/stores/modules/myApps.store';
  import { useEventStore } from '@/stores/event.store';

  export default {
    name: 'Apps',
    components: { AppGrid, EmptyApps },
    data() {
      return {
        searchTerm: '',
        eventStore: useEventStore(),
      };
    },
    /* istanbul ignore next */
    mounted() {
      this.fetchCategories();
      this.on('updateGrid', this.fetchCategories);
    },
    beforeUnmount() {
      this.off('updateGrid', this.fetchCategories);
    },
    /* istanbul ignore next */
    computed: {
      ...mapState(auth_store, ['project']),
      ...mapState(my_apps, [
        'configuredApps',
        'errorConfiguredApps',
        'errorConfiguredApps',
        'installedApps',
        'loadingInstalledApps',
        'errorInstalledApps',
      ]),
      hasApps: function () {
        if (this.loadingConfiguredApps || this.loadingInstalledApps) {
          return true;
        }

        return this.configuredApps?.length || this.installedApps?.length;
      },
      searchOptions() {
        if (!this.configuredApps || !this.installedApps) return [];

        const allApps = [...this.configuredApps, ...this.installedApps];

        const filtered = this.filterByName(allApps, this.searchTerm);

        return filtered.map((app) => {
          let name = app.generic ? app.config.channel_name : app.name;
          if (app.config?.title) {
            name += ` - ${app.config?.title}`;
          }
          return name;
        });
      },
      filteredConfiguredApps() {
        if (!this.configuredApps) return [];

        if (!this.searchTerm || !this.searchTerm.trim()) return this.configuredApps;

        return this.filterByName(this.configuredApps, this.searchTerm);
      },
      filteredInstalledApps() {
        if (!this.installedApps) return [];

        if (!this.searchTerm || !this.searchTerm.trim()) return this.installedApps;

        return this.filterByName(this.installedApps, this.searchTerm);
      },
    },
    methods: {
      ...mapActions(my_apps, ['getConfiguredApps', 'getInstalledApps']),
      ...mapActions(useEventStore, ['on', 'off']),
      filterByName(appList, search) {
        return appList.filter((app) => {
          const appMainName = app.generic ? app.config.channel_name : app.name;
          const appName = app.config?.title || app.config?.channel_name;

          const name = appMainName + appName;
          return name.toLowerCase().includes(search.toLowerCase());
        });
      },
      fetchCategories() {
        this.fetchConfigured();
        this.fetchInstalled();
      },
      fetchConfigured() {
        const params = {
          project_uuid: this.project,
        };
        this.getConfiguredApps({ params });

        if (this.errorConfiguredApps) {
          this.callErrorModal({ text: this.$t('apps.myApps.error.configured') });
        }
      },
      fetchInstalled() {
        const params = {
          project_uuid: this.project,
        };
        this.getInstalledApps({ params });

        if (this.errorInstalledApps) {
          this.callErrorModal({ text: this.$t('apps.myApps.error.installed') });
        }
      },
      navigateToDiscovery() {
        this.$router.replace('/apps/discovery');
      },
      callErrorModal({ text }) {
        unnnic.unnnicCallAlert({
          props: {
            text: text,
            type: 'error',
          },
          seconds: 6,
        });
      },
      handleEvent() {
        console.log('chegou');
        this.fetchCategories();
      },
    },
  };
</script>

<style lang="scss" scoped>
  @import './styles.scss';
</style>
