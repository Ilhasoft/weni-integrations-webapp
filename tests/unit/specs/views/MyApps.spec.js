import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import MyApps from '@/views/MyApps.vue';
import AppGrid from '@/components/AppGrid.vue';
import { singleApp } from '../../../__mocks__/appMock';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('MyApps.vue', () => {
  let wrapper;
  let actions;
  let getters;
  let store;

  beforeEach(() => {
    // const fetchCategoriesSpy = spyOn(MyApps.methods, 'fetchCategories');
    actions = {
      getConfiguredApps: jest.fn(() => {
        return { data: [singleApp] };
      }),
      getInstalledApps: jest.fn(() => {
        return { data: [singleApp] };
      }),
    };

    getters = {
      getSelectedProject: jest.fn(() => {
        return '123';
      }),
    };

    store = new Vuex.Store({
      actions,
      getters,
    });

    wrapper = shallowMount(MyApps, {
      localVue,
      store,
      mocks: {
        $t: () => 'some specific text',
        $router: {
          replace: jest.fn(),
        },
        $route: {
          path: '/apps/1/details',
        },
      },
      stubs: {
        AppGrid,
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('hasApps', () => {
    it('should return true if loadings are still ocurring', () => {
      expect(wrapper.vm.hasApps).toBeTruthy();
    });

    it('should return true if not loading and have data', async () => {
      await wrapper.setData({
        configuredApps: {
          loading: false,
          data: [singleApp],
        },
      });
      await wrapper.setData({
        installedApps: {
          loading: false,
          data: [],
        },
      });

      expect(wrapper.vm.hasApps).toBeTruthy();
    });

    it('should return false if not loading and dont have data', async () => {
      await wrapper.setData({
        configuredApps: {
          loading: false,
          data: [],
        },
      });
      await wrapper.setData({
        installedApps: {
          loading: false,
          data: [],
        },
      });

      expect(wrapper.vm.hasApps).toBeFalsy();
    });
  });

  describe('navigateToDiscovery', () => {
    it('should change route to discovery', () => {
      const spy = spyOn(wrapper.vm.$router, 'replace');
      wrapper.vm.navigateToDiscovery();

      expect(spy).toBeCalledTimes(1);
      expect(spy).toHaveBeenCalledWith(`/apps/discovery`);
    });
  });

  describe('fetchCategories()', () => {
    it('should call fetchConfigured()', async () => {
      const spy = spyOn(wrapper.vm, 'fetchConfigured');
      expect(spy).not.toHaveBeenCalled();
      await wrapper.vm.fetchCategories();
      expect(spy).toHaveBeenCalledTimes(1);
    });

    it('should call fetchInstalled()', async () => {
      const spy = spyOn(wrapper.vm, 'fetchInstalled');
      expect(spy).not.toHaveBeenCalled();
      await wrapper.vm.fetchCategories();
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });

  describe('fetchConfigured()', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('should call getConfiguredApps()', async () => {
      expect(actions.getConfiguredApps).not.toHaveBeenCalled();
      await wrapper.vm.fetchConfigured();
      expect(actions.getConfiguredApps).toHaveBeenCalledTimes(1);
    });

    it('should set new data on configuredApps', async () => {
      await wrapper.setData({ configuredApps: { loading: true, data: null } });
      expect(wrapper.vm.configuredApps.data).toBeNull();
      await wrapper.vm.fetchConfigured();
      expect(wrapper.vm.configuredApps.data).toEqual([singleApp]);
    });
  });

  describe('fetchInstalled()', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('should call getInstalledApps()', async () => {
      expect(actions.getInstalledApps).not.toHaveBeenCalled();
      await wrapper.vm.fetchInstalled();
      expect(actions.getInstalledApps).toHaveBeenCalledTimes(1);
    });

    it('should set new data on installedApps', async () => {
      await wrapper.setData({ installedApps: { loading: true, data: null } });
      expect(wrapper.vm.installedApps.data).toBeNull();
      await wrapper.vm.fetchInstalled();
      expect(wrapper.vm.installedApps.data).toEqual([singleApp]);
    });
  });
});
