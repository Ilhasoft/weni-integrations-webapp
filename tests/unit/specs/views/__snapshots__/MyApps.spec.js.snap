// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyApps.vue should be rendered properly 1`] = `
<div class="my-apps">
  <unnnic-autocomplete-stub type="normal" data="Weni Web Chat - Chat Title,A random generic,Weni Web Chat - Chat Title,A random generic" value="" iconleft="search-1" tooltipsideiconright="right" size="md" placeholder="some specific text" class="my-apps__search"></unnnic-autocomplete-stub>
  <!---->
  <div class="my-apps__sections">
    <div class="my-apps__sections__grids">
      <div>
        <section id="app-grid">
          <div class="app-grid__header">
            <unnnic-avatar-icon-stub enabled="true" icon="cog-1" size="sm" scheme="aux-purple"></unnnic-avatar-icon-stub>
            <p class="app-grid__header__title">some specific text</p>
          </div>
          <div class="app-grid__content">
            <unnnic-card-stub type="marketplace" title="Weni Web Chat - Chat Title" description="some specific text" percent="0" rating="4.9" comments="undefined some specific text" enabled="true" iconsrc="https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg" infoposition="right" hasinformationicon="true" scheme="aux-blue" clickable="true" typeaction="edit" buttonaction="[Function]" class="app-grid__content__item">
              <unnnic-dropdown-stub position="bottom-left" class="app-grid__content__item__dropdown">
                <unnnic-button-stub size="small" type="tertiary" iconcenter="navigation-menu-vertical-1"></unnnic-button-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--action">
                  <unnnic-icon-svg-stub icon="pencil-write-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  Edit
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--details">
                  <unnnic-icon-svg-stub icon="export-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--details">
                  <unnnic-icon-svg-stub icon="copy-paste-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--remove">
                  <unnnic-icon-svg-stub icon="bin-1-1" size="sm" scheme="feedback-red"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
              </unnnic-dropdown-stub>
            </unnnic-card-stub>
            <unnnic-card-stub type="marketplace" title="A random generic - undefined" description="Ullamco occaecat et id cillum." percent="0" rating="4.9" comments="undefined some specific text" enabled="true" infoposition="right" hasinformationicon="true" scheme="aux-blue" clickable="true" typeaction="edit" buttonaction="[Function]" class="app-grid__content__item">
              <unnnic-dropdown-stub position="bottom-left" class="app-grid__content__item__dropdown">
                <unnnic-button-stub size="small" type="tertiary" iconcenter="navigation-menu-vertical-1"></unnnic-button-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--action">
                  <unnnic-icon-svg-stub icon="pencil-write-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  Edit
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--details">
                  <unnnic-icon-svg-stub icon="export-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--details">
                  <unnnic-icon-svg-stub icon="copy-paste-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--remove">
                  <unnnic-icon-svg-stub icon="bin-1-1" size="sm" scheme="feedback-red"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
              </unnnic-dropdown-stub>
            </unnnic-card-stub>
          </div>
          <!---->
        </section>
        <unnnic-modal-stub text="some specific text" modalicon="alert-circle-1" closeicon="true" scheme="feedback-red"><span>some specific text</span>
          <unnnic-button-stub size="large" type="tertiary">some specific text</unnnic-button-stub>
          <loadingbutton-stub type="primary" size="large" loadingtext="some specific text" loadingposition="left" text="some specific text"></loadingbutton-stub>
        </unnnic-modal-stub>
        <config-modal-stub></config-modal-stub>
      </div>
      <div>
        <section id="app-grid">
          <div class="app-grid__header">
            <unnnic-avatar-icon-stub enabled="true" icon="check-circle-1-1" size="sm" scheme="aux-blue"></unnnic-avatar-icon-stub>
            <p class="app-grid__header__title">some specific text</p>
          </div>
          <div class="app-grid__content">
            <unnnic-card-stub type="marketplace" title="Weni Web Chat" description="some specific text" percent="0" rating="4.9" comments="undefined some specific text" enabled="true" iconsrc="https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg" infoposition="right" hasinformationicon="true" scheme="aux-blue" clickable="true" typeaction="config" buttonaction="[Function]" class="app-grid__content__item">
              <unnnic-dropdown-stub position="bottom-left" class="app-grid__content__item__dropdown">
                <unnnic-button-stub size="small" type="tertiary" iconcenter="navigation-menu-vertical-1"></unnnic-button-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--action">
                  <unnnic-icon-svg-stub icon="cog-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  Configure
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--details">
                  <unnnic-icon-svg-stub icon="export-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--details">
                  <unnnic-icon-svg-stub icon="copy-paste-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--remove">
                  <unnnic-icon-svg-stub icon="bin-1-1" size="sm" scheme="feedback-red"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
              </unnnic-dropdown-stub>
            </unnnic-card-stub>
            <unnnic-card-stub type="marketplace" title="A random generic" description="Ullamco occaecat et id cillum." percent="0" rating="4.9" comments="undefined some specific text" enabled="true" infoposition="right" hasinformationicon="true" scheme="aux-blue" clickable="true" typeaction="config" buttonaction="[Function]" class="app-grid__content__item">
              <unnnic-dropdown-stub position="bottom-left" class="app-grid__content__item__dropdown">
                <unnnic-button-stub size="small" type="tertiary" iconcenter="navigation-menu-vertical-1"></unnnic-button-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--action">
                  <unnnic-icon-svg-stub icon="cog-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  Configure
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--details">
                  <unnnic-icon-svg-stub icon="export-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--details">
                  <unnnic-icon-svg-stub icon="copy-paste-1" size="sm" scheme="neutral-darkest"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
                <unnnic-dropdown-item-stub class="app-grid__content__item__button--remove">
                  <unnnic-icon-svg-stub icon="bin-1-1" size="sm" scheme="feedback-red"></unnnic-icon-svg-stub>
                  some specific text
                </unnnic-dropdown-item-stub>
              </unnnic-dropdown-stub>
            </unnnic-card-stub>
          </div>
          <!---->
        </section>
        <unnnic-modal-stub text="some specific text" modalicon="alert-circle-1" closeicon="true" scheme="feedback-red"><span>some specific text</span>
          <unnnic-button-stub size="large" type="tertiary">some specific text</unnnic-button-stub>
          <loadingbutton-stub type="primary" size="large" loadingtext="some specific text" loadingposition="left" text="some specific text"></loadingbutton-stub>
        </unnnic-modal-stub>
        <config-modal-stub></config-modal-stub>
      </div>
    </div>
  </div>
</div>
`;
